modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -386,6 +388,11 @@ public static String getSimpleRef(String ref) {
 
         }
 
+        try {
+            ref = URLDecoder.decode(ref, "UTF-8");
+        } catch (UnsupportedEncodingException ignored) {
+        }
+
```

Comment:
Shall we decode it at the start of the function `getSimpleRef` instead?

Reply:
I had the same thought, but why decoding a String part which gets thrown away? I see no reason to do so

By the way, I think we should also unescape these special cases:
![image](https://user-images.githubusercontent.com/11720038/91284800-af420a80-e78c-11ea-947d-02c1072f3afd.png)